### PR TITLE
cppcheck: update and support rule-file.

### PR DESCRIPTION
**Pull request by layus**
- [X] Tested on NixOS.

### CODE REVIEW

File: pkgs/development/tools/analysis/cppcheck/default.nix [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
   name = "${pname}-${version}";
 
   src = fetchurl {
     url = "mirror://sourceforge/${pname}/${name}.tar.bz2";
-    sha256 = "1l46bmzm5syfr9m5l0bqkj8lcyrynhw8gjf95s4fwhp2b7f0zisv";
+    sha256 = "1cc60y2vjq4g88183jkan91ybzqy0n6p5ncs8z5rh6rjsvvrc9p4";
   };
 
-  nativeBuildInputs = [ libxslt docbook_xsl docbook_xml_dtd_45 ];
+  nativeBuildInputs = [ libxslt docbook_xsl docbook_xml_dtd_45 pcre ];
```
Comment (joachifm) [inline pkgs/development/tools/analysis/cppcheck/default.nix]: Does this mean cppcheck doesn't use pcre at runtime for this?

Comment (layus) [inline pkgs/development/tools/analysis/cppcheck/default.nix]: Well, no :-). Good catch. That being said, there is no real difference in practice :-D.

**Comment by layus** [comment]
@joachifm Updated according to comment.

**Comment by Mic92** [comment]
seems to fail now on darwin: https://hydra.nixos.org/build/51724675

**Pull request merged**